File: 项目/OC项目/百思不得姐(oc)/百思不得姐(oc)/Classes/Essence(精华)/View/CommentCell.m
//
//  CommentCell.m
//  百思不得姐(oc)
//
//  Created by 金亮齐 on 2017/5/16.
//  Copyright © 2017年 醉看红尘这场梦. All rights reserved.
//

#import "CommentCell.h"
#import "UIImageView+WebCache.h"
#import "TopCommentItem.h"
#import "UserItems.h"
@interface CommentCell()

@property (weak, nonatomic) IBOutlet UIImageView *iconView;

@property (weak, nonatomic) IBOutlet UIImageView *SexImage;

@property (weak, nonatomic) IBOutlet UILabel *nameView;

@property (weak, nonatomic) IBOutlet UILabel *textView;

@property (weak, nonatomic) IBOutlet UILabel *likeCountView;

@end
@implementation CommentCell

- (BOOL)canBecomeFirstResponder
{
    return YES;
}

- (BOOL)canPerformAction:(SEL)action withSender:(id)sender
{
    return NO;
}
- (void)awakeFromNib {
    [super awakeFromNib];
    self.iconView.layer.cornerRadius = (self.iconView.sy_width * 0.5);
    self.iconView.layer.masksToBounds = YES;

}

-(void)setTopComment:(TopCommentItem *)topComment
{
    _topComment = topComment;
    [self.iconView sd_setImageWithURL:[NSURL URLWithString:topComment.user.profile_image] placeholderImage:[UIImage imageNamed:@"defaultTagIcon"]];
    self.SexImage.image = [topComment.user.sex isEqualToString:@"m"] ? [UIImage imageNamed:@"Profile_manIcon"] : [UIImage imageNamed:@"Profile_womanIcon"];
    self.textView.text = topComment.content;
    self.nameView.text = topComment.user.username;
    self.likeCountView.text = topComment.like_count;
}
@end

File: 项目/Swift项目/百思不得姐/百思不得姐/Classes/Essence(精华)/View/CommentCell.swift
//
//  CommentCell.swift
//  百思不得姐
//
//  Created by 金亮齐 on 2017/11/14.
//  Copyright © 2017年 金亮齐. All rights reserved.
//

import UIKit
import Kingfisher

class CommentCell: UITableViewCell {
    
    static let commentCellIdentifier: String = "commentCell"
    
    static func commentCell(_ tableView: UITableView) -> CommentCell {
        
        var cell = tableView.dequeueReusableCell(withIdentifier: CommentCell.commentCellIdentifier)
        
        if cell == nil {
            cell = Bundle.main.loadNibNamed("CommentCell", owner: nil, options: nil)?.last as? UITableViewCell
        }
        return cell as! CommentCell
    }
    
    @IBOutlet weak var headerView: UIImageView!
    
    @IBOutlet weak var sexView: UIImageView!
    
    @IBOutlet weak var nameLabel: UILabel!
    
    @IBOutlet weak var contentLabel: UILabel!
    
    @IBOutlet weak var zaiCountLabel: UILabel!
    
    public func setupUI(_ comment: Comment) {
        
        headerView.setHeaderImage(comment.user.profile_image)
        
        sexView.image = comment.user.sex == "m" ? UIImage(named: "Profile_manIcon") : UIImage(named: "Profile_womanIcon")
        
        nameLabel.text = comment.user.username
        
        contentLabel.text = comment.content
        
        zaiCountLabel.text = "\(comment.like_count)"
    }
    
    override func awakeFromNib() {
        super.awakeFromNib()
    }
    
    // MARK: - 设置为true 让自身成为第一响应者
    override var canBecomeFirstResponder: Bool {
        return true
    }
    // MARK: - 设置为false 不使用系统自带的item
    override func canPerformAction(_ action: Selector, withSender sender: Any?) -> Bool {
        return false
    }
}
